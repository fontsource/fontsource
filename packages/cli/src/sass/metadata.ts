import type { Axes, Metadata, UnicodeRange } from '../types';
import { findClosest } from '../utils';

const axesValue = (metadata: Metadata) => {
	let out = '(\n';
	for (const variable of Object.keys(metadata.variable)) {
		const axis = (metadata.variable as Axes)[variable];
		out += `  ${variable}: (
    default: ${axis.default},
    min: ${axis.min},
    max: ${axis.max},
    step: ${axis.step},
  ),
`;
	}
	out += ')';
	return out;
};

const unicodeValue = (unicode: UnicodeRange) => {
	if (Object.keys(unicode).length > 0) {
		let out = '(\n';

		for (const unicodeKey of Object.keys(unicode)) {
			const unicodeVal = unicode[unicodeKey];

			out += `  ${unicodeKey
				.replace('[', '')
				.replace(']', '')}: (${unicodeVal}),\n`;
		}

		out += ')';

		return out;
	}

	return 'null';
};

const defaultAxis = (metadata: Metadata) => {
	const axes = Object.keys(metadata.variable);
	if (axes.includes('wght')) return 'wght';
	if (axes.includes('full')) return 'full';
	return axes[0];
};

export const sassMetadata = (
	metadata: Metadata,
	unicode: UnicodeRange,
	isVariable: boolean,
) => {
	return `$metadata: (
  id: '${metadata.id}',
  family: '${metadata.family}',
  category: ${metadata.category},
  subsets: (${metadata.subsets.join(', ')}),
  weights: (${metadata.weights.join(', ')}),
  styles: (${metadata.styles.join(', ')}),
  axes: ${isVariable ? axesValue(metadata) : 'null'},
  defaults: (
    subset: ${metadata.defSubset},
    weight: ${findClosest(metadata.weights, 400)},
    style: normal,
    axis: ${isVariable ? defaultAxis(metadata) : 'null'},
  ),
  unicode: ${unicodeValue(unicode)}
) !default;`;
};
