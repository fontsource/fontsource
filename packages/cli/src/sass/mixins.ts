export const sassMixins = `@use 'sass:list';
@use 'sass:map';
@use 'sass:math';
@use 'sass:meta';
@use 'sass:string';

@use 'metadata';

$metadata: meta.module-variables(metadata) !default;
$directory: null !default;

$family: null !default;
$display: null !default;
$displayVar: null !default;
$formats: null !default;
$subsets: null !default;
$weights: null !default;
$styles: null !default;
$axes: null !default;

@mixin generator(
  $metadata: $metadata,
  $directory: $directory,
  $family: $family,
  $display: $display,
  $displayVar: $displayVar,
  $formats: $formats,
  $subsets: $subsets,
  $weights: $weights,
  $styles: $styles,
  $axes: $axes
) {
  $isVariable: map.get($metadata, axes) != null;

  $directory: if(
    $directory,
    $directory,
    '~@fontsource#{if($isVariable, '-variable', '')}/#{map.get($metadata, id)}/files'
  );

  $family: if($family, $family, map.get($metadata, family) + if($isVariable, ' Variable', ''));
  $display: if($display, $display, swap);
  $displayVar: if($displayVar != null, $displayVar, true);
  $formats: if(not $formats or $formats == all, if($isVariable, woff2, (woff2, woff)), $formats);
  $subsets: if(
    $subsets,
    if($subsets == all, map.get($metadata, subsets), $subsets),
    map.get($metadata, defaults, subset)
  );
  $weights: if(
    $weights,
    if($weights == all, map.get($metadata, weights), $weights),
    map.get($metadata, defaults, weight)
  );
  $styles: if(
    $styles,
    if($styles == all, map.get($metadata, styles), $styles),
    map.get($metadata, defaults, style)
  );
  $axes: if(
    $axes,
    if($axes == all, full, $axes),
    if($isVariable, if(map.has-key($metadata, axes, wght), wght, full), null)
  );

  @each $subset in $subsets {
    @each $unicodeSubset, $unicodeRange in map.get($metadata, unicode) {
      @if (
        ($subset == $unicodeSubset) or
          (
            // Is numeric subset
            ($subset == map.get($metadata, defaults, subset)) and not
              list.index(map.get($metadata, subsets), $unicodeSubset)
          )
      ) {
        @each $weight in if($axes, null, $weights) {
          @each $axis in $axes {
            @each $style in $styles {
              $variant: '#{map.get($metadata, id)}-#{if($unicodeSubset, $unicodeSubset, $subset)}-#{if($axis, $axis, $weight)}-#{$style}';

              $src: ();
              @each $format in $formats {
                $src: append(
                  $src,
                  url('#{$directory}/#{$variant}.#{$format}')
                    format('#{$format}#{if($axis, '-variations', '')}'),
                  comma
                );
              }

              @content ((
                metadata: $metadata,
                directory: $directory,
                family: $family,
                display: $display,
                displayVar: $displayVar,
                formats: $formats,
                subsets: $subsets,
                weights: $weights,
                styles: $styles,
                axes: $axes,

                variant: $variant,
                subset: $subset,
                unicodeSubset: $unicodeSubset,
                unicodeRange: $unicodeRange,
                weight: $weight,
                axis: $axis,
                style: $style,

                font-family: string.quote($family),
                font-style: if(
                  (($axis == full) or ($axis == slnt)) and map.has-key($metadata, axes, slnt),
                  oblique map.get($metadata, axes, slnt, min) + deg map.get($metadata, axes, slnt, max) + deg,
                  $style
                ),
                font-display: if($displayVar, var(--fontsource-display, $display), $display),
                font-weight: if(
                  (($axis == full) or ($axis == wght)) and map.has-key($metadata, axes, wght),
                  map.get($metadata, axes, wght, min) map.get($metadata, axes, wght, max),
                  $weight
                ),
                font-stretch: if(
                  (($axis == full) or ($axis == wdth)) and map.has-key($metadata, axes, wdth),
                  '#{map.get($metadata, axes, wdth, min)}% #{map.get($metadata, axes, wdth, max)}%',
                  null
                ),
                src: $src,
                unicode-range: $unicodeRange,
              ));
            }
          }
        }
      }
    }
  }
}

@mixin faces(
  $metadata: $metadata,
  $directory: $directory,
  $family: $family,
  $display: $display,
  $displayVar: $displayVar,
  $formats: $formats,
  $subsets: $subsets,
  $weights: $weights,
  $styles: $styles,
  $axes: $axes
) {
  @include generator(
      $metadata: $metadata,
      $directory: $directory,
      $family: $family,
      $display: $display,
      $displayVar: $displayVar,
      $formats: $formats,
      $subsets: $subsets,
      $weights: $weights,
      $styles: $styles,
      $axes: $axes
    )
    using ($props) {
    /* #{map.get($props, variant)} */
    @font-face {
      font-family: map.get($props, font-family);
      font-style: map.get($props, font-style);
      font-display: map.get($props, font-display);
      font-weight: map.get($props, font-weight);
      font-stretch: map.get($props, font-stretch);
      unicode-range: map.get($props, unicode-range);
      src: map.get($props, src);
    }
  }
}
`;
