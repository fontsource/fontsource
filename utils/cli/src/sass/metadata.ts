import { Axes, Metadata, UnicodeRange } from '../types';
import { findClosest, sassVar } from '../utils';

const subsetValue = (metadata: Metadata, unicodeRange: UnicodeRange) => {
	// Merge metadata subsets and unicode range
	const subsets: Record<string, string[]> = {};
	for (const subset of metadata.subsets) {
		subsets[subset] = [];
	}
	for (const range of Object.keys(unicodeRange)) {
		const rangeNumMatch = range.match(/\[d+]/);
		if (rangeNumMatch && rangeNumMatch.length > 0) {
			// Cover numeric unicode ranges
			subsets[metadata.defSubset] ??= [];
			subsets[metadata.defSubset][Number.parseInt(rangeNumMatch[0], 10)] =
				unicodeRange[range];
		} else {
			subsets[range] = [unicodeRange[range]];
		}
	}

	// Generate sass map from subsets
	let subsetsValue = '(\n';
	for (const subset of Object.keys(subsets)) {
		subsetsValue += `  ${subset}: `;

		if (subsets[subset].length > 0) {
			subsetsValue += '(\n';
			for (const range of subsets[subset]) {
				subsetsValue += `    (${range}),\n`;
			}
			subsetsValue += '  ),\n';
		} else {
			subsetsValue += 'null';
		}

		subsetsValue += ',\n';
	}
	subsetsValue += ')';

	return subsetsValue;
};

const axesValue = (metadata: Metadata) => {
	if (metadata.variable && Object.keys(metadata.variable).length > 0) {
		let subsetsValue = '(\n';
		for (const variable of Object.keys(metadata.variable)) {
			const axis = (metadata.variable as Axes)[variable];
			subsetsValue += `  ${variable}: (
    default: ${axis.default},
    min: ${axis.min},
    max: ${axis.max},
    step: ${axis.step},
  ),
`;
		}
		subsetsValue += ')';

		return subsetsValue;
	}

	return '()';
};

export const sassMetadata = (
	metadata: Metadata,
	unicodeRange: UnicodeRange
) => {
	let out = '';
	out += sassVar('id', `'${metadata.id}'`);
	out += sassVar('family', `'${metadata.family}'`);
	out += sassVar('category', `${metadata.category}`);
	out += sassVar('weights', `(${metadata.weights.join(', ')})`);
	out += sassVar('styles', `(${metadata.styles.join(', ')})`);
	out += sassVar(
		'defaults',
		`(
  subset: ${metadata.defSubset},
  weight: ${findClosest(metadata.weights, 400)},
  style: normal,
)`
	);
	out += sassVar('subsets', subsetValue(metadata, unicodeRange));
	out += sassVar('axes', axesValue(metadata));

	return out;
};
