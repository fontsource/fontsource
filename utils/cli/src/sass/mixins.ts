export const sassMixins = `@use 'sass:list';
@use 'sass:map';
@use 'sass:math';
@use 'sass:meta';
@use 'sass:string';

@use 'metadata';

$metadata: meta.module-variables(metadata) !default;
$directory: null !default;

$family: null !default;
$display: null !default;
$displayVar: null !default;
$formats: null !default;
$subsets: null !default;
$weights: null !default;
$styles: null !default;
$axes: null !default;

@mixin generator(
  $metadata: $metadata,
  $directory: $directory,
  $family: $family,
  $display: $display,
  $displayVar: $displayVar,
  $formats: $formats,
  $subsets: $subsets,
  $weights: $weights,
  $styles: $styles,
  $axes: $axes
) {
  $directory: if(
    $directory,
    $directory,
    '~@fontsource/#{map.get($metadata, id)}/files'
  );

  $family: if($family, $family, map.get($metadata, family));
  $display: if($display, $display, swap);
  $displayVar: if($displayVar != null, $displayVar, true);
  $formats: if(not $formats or $formats == all, (woff2, woff), $formats);
  $subsets: if(
    $subsets,
    if($subsets == all, map.get($metadata, subsets), $subsets),
    map.get($metadata, defaults, subset)
  );
  $weights: if(
    $weights,
    if($weights == all, map.get($metadata, weights), $weights),
    map.get($metadata, defaults, weight)
  );
  $styles: if(
    $styles,
    if($styles == all, map.get($metadata, styles), $styles),
    map.get($metadata, defaults, style)
  );
  $axes: if(
    $axes,
    if($axes == all, full, $axes),
    if(
      map.get($metadata, axes),
      if(map.has-key($metadata, axes, wght), wght, full),
      null
    )
  );

  @each $subset in $subsets {
    @each $unicodeSubset, $unicodeRange in map.get($metadata, unicode) {
      @if (
        $subset ==
          $unicodeSubset or
          (
            // Is numeric subset
            $subset ==
              map.get($metadata, defaults, subset) and not
              map.has-key($metadata, subsets, $unicodeSubset)
          )
      ) {
        @each $weight in if($axes, null, $weights) {
          @each $axis in $axes {
            @each $style in $styles {
              $variant: '#{map.get($metadata, id)}-#{$unicodeSubset}-#{if($axis, $axis, $weight)}-#{$style}';

              $src: ();
              @each $format in $formats {
                $src: append(
                  $src,
                  url('#{$directory}/#{$variant}.#{$format}')
                    format('#{$format}'),
                  comma
                );
              }

              @content ((
                metadata: $metadata,
                directory: $directory,
                family: $family,
                display: $display,
                displayVar: $displayVar,
                formats: $formats,
                subsets: $subsets,
                weights: $weights,
                styles: $styles,
                axes: $axes,

                variant: $variant,
                subset: $subset,
                unicodeSubset: $unicodeSubset,
                unicodeRange: $unicodeRange,
                weight: $weight,
                axis: $axis,
                style: $style,

                font-family: string.quote($family),
                font-style: if(
                  ($axis == full or $axis == slnt) and map.has-key($metadata, axes, slnt), 
                  oblique map.get($metadata, axes, slnt, min) + deg map.get($metadata, axes, slnt, max) + deg,
                  $style
                ),
                font-display: if($displayVar, var(--fontsource-display, $display), $display),
                font-weight: if(
                  ($axis == full or $axis == wght) and map.has-key($metadata, axes, wght), 
                  map.get($metadata, axes, wght, min) map.get($metadata, axes, wght, max),
                  $weight
                ),
                font-stretch: if(
                  ($axis == full or $axis == wdth) and map.has-key($metadata, axes, wdth), 
                  '#{map.get($metadata, axes, wdth, min)}% #{map.get($metadata, axes, wdth, max)}%',
                  null
                ),
                src: $src,
                unicode-range: $unicodeRange,
              ));
            }
          }
        }
      }
    }
  }
}

@mixin faces(
  $metadata: $metadata,
  $directory: $directory,
  $family: $family,
  $display: $display,
  $displayVar: $displayVar,
  $formats: $formats,
  $subsets: $subsets,
  $weights: $weights,
  $styles: $styles,
  $axes: $axes
) {
  @include generator(
      $metadata: $metadata,
      $directory: $directory,
      $family: $family,
      $display: $display,
      $displayVar: $displayVar,
      $formats: $formats,
      $subsets: $subsets,
      $weights: $weights,
      $styles: $styles,
      $axes: $axes
    )
    using ($data) {
    /* #{map.get($data, variant)} */
    @font-face {
      font-family: map.get($data, font-family);
      font-style: map.get($data, font-style);
      font-display: map.get($data, font-display);
      font-weight: map.get($data, font-weight);
      font-stretch: map.get($data, font-stretch);
      unicode-range: map.get($data, unicode-range);
      src: map.get($data, src);

      @content();
    }
  }
}
`;
