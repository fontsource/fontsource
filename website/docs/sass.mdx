---
title: Hello World
---

This is an example post. There's another one [here](/docs/sass).