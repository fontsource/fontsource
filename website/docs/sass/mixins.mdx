---
title: Mixins
---

# Sass Mixins

---

A basic implementation into your typical project that compiles with Sass/SCSS is the following:

```scss
@import "~@fontsource/open-sans/index.css";
@import "~@fontsource/open-sans/300-italic.css";
```

## @use

---

For more advanced setups, you can use our highly customizable Sass mixins that can modify many of the existing [`@font-face`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face) variables.

You can view what the full SCSS file looks like [here](https://github.com/fontsource/fontsource/tree/master/packages/open-sans/scss/mixins.scss).

```scss
@use "@fontsource/open-sans/scss/mixins" as OpenSans;
```

> **Note:** This uses the [`@use`](https://sass-lang.com/documentation/at-rules/use) Sass function which is only supported by Dart Sass. For older, deprecated solutions read over the Legacy section below.

### FontFace Mixin

---

Generates a CSS file similar to [this](https://github.com/fontsource/fontsource/blob/main/packages/open-sans/400.css) where all unicode-range subsets are generated using the [`@each`](https://sass-lang.com/documentation/at-rules/control/each) Sass function.

```scss
@include OpenSans.fontFace(
  $weight: 500,
  $display: fallback,
  $fontDir: "~@fontsource/open-sans/files"
);
```

### FontFace Mixin Inputs

---

Only supported by Google fonts. For fonts outside that ecosystem, only `fontFaceCustom` is usable.

| Variable    | CSS Selector                                                                                 | Options                                         | Default                           |
| ----------- | -------------------------------------------------------------------------------------------- | ----------------------------------------------- | --------------------------------- |
| $weight     | [`font-weight`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight)                | Varies per font package.                        | `400`                             |
| $style      | [`font-style`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-style)                  | `normal`, `italic`                              | `normal`                          |
| $display    | [`font-display`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display)   | `auto`, `block`, `swap`, `fallback`, `optional` | `swap`                            |
| $fontName   | [`font-family`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family)                | Up to user preference.                          | Varies per font package.          |
| $fontId     | N/A                                                                                          | Used to resolve default $fontDir.               | Varies per font package.          |
| $fontDir    | `src`                                                                                        | Directory where all font files are.             | `"~@fontsource/#{$fontId}/files"` |
| $unicodeMap | [`unicode-range`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/unicode-range) | Key value pair of subsets and unicode values.   | Varies per font package.          |

### FontFaceCustom Mixin

---

For a far more granular mixin that removes the [`@each`](https://sass-lang.com/documentation/at-rules/control/each) function and allows the developer to specify additional parameters such as including unicode-range or choosing different paths for either the `woff` or `woff2` file.

```scss
@include OpenSans.fontFaceCustom(
  $weight: 600,
  $display: optional,
  $woff2Path: "~@fontsource/open-sans/files/open-sans-latin-500-normal.woff2",
  $unicodeRange: false
);
```

### FontFaceCustom Mixin Inputs

---

Included in both Google and other font foundry families.

| Variable            | CSS Selector                                                                                 | Options                                         | Default                                                                       |
| ------------------- | -------------------------------------------------------------------------------------------- | ----------------------------------------------- | ----------------------------------------------------------------------------- |
| $weight             | [`font-weight`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight)                | Varies per font package.                        | `400`                                                                         |
| $style              | [`font-style`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-style)                  | `normal`, `italic`                              | `normal`                                                                      |
| $display            | [`font-display`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display)   | `auto`, `block`, `swap`, `fallback`, `optional` | `swap`                                                                        |
| $fontName           | [`font-family`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family)                | Up to user preference.                          | Varies per font package.                                                      |
| $fontId             | N/A                                                                                          | Used to resolve default font file paths.        | Varies per font package.                                                      |
| $woff2Path          | `src`                                                                                        | Direct filepath to WOFF2 font file.             | `"~@fontsource/#{$fontId}/files/#{$fontId}-latin-#{$weight}-#{$style}.woff2"` |
| $woffPath           | `src`                                                                                        | Direct filepath to WOFF font file.              | `"~@fontsource/#{$fontId}/files/#{$fontId}-latin-#{$weight}-#{$style}.woff"`  |
| $unicodeRange       | [`unicode-range`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/unicode-range) | `true`, `false`                                 | `false`                                                                       |
| $unicodeRangeValues | [`unicode-range`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/unicode-range) | Values for unicode-range.                       | Latin unicode range.                                                          |

---

## Legacy Sass Implementations

---

[`@use`](https://sass-lang.com/documentation/at-rules/use) is only compatible with Dart Sass. [`@import`](https://sass-lang.com/documentation/at-rules/import) is for all other deprecated Sass implementations.

```scss
@import "~@fontsource/open-sans/scss/mixins";

@include fontFace(
  $weight: 500,
  $display: fallback,
  $fontDir: "~@fontsource/open-sans/files"
);

@include fontFaceCustom(
  $weight: 600,
  $display: optional,
  $woff2Path: "~@fontsource/open-sans/files/open-sans-latin-500-normal.woff2",
  $unicodeRange: false
);
```

> **Note:** This implementation can be problematic as the mixin variables may pollute the global namespace and affect other components in your project.
