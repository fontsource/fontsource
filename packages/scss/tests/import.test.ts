import { compileString, NodePackageImporter } from 'sass';
import { describe, expect, it } from 'vitest';

const compileSass = (path: string) => {
	const res = compileString(`@use ${path};`, {
		importers: [new NodePackageImporter()],
	});

	return res.css.toString();
};

describe('scss imports', () => {
	it('should import with default path', async () => {
		expect(
			compileSass("'pkg:@fontsource/carlito/index.css'"),
		).toMatchSnapshot();
	});

	it('should import with default and namespace', async () => {
		expect(
			compileSass("'pkg:@fontsource/carlito/index.css' as carlito"),
		).toMatchSnapshot();
	});

	it('should import with weight path and extension', async () => {
		expect(
			compileSass("'pkg:@fontsource/carlito/400.css' as carlito"),
		).toMatchSnapshot();
	});
});
