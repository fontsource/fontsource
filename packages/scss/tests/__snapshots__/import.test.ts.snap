// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`scss imports > should import with default and namespace 1`] = `
"/* carlito-cyrillic-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-cyrillic-ext-400-normal.woff2) format("woff2"), url(./files/carlito-cyrillic-ext-400-normal.woff) format("woff");
  unicode-range: U+0460-052F, U+1C80-1C8A, U+20B4, U+2DE0-2DFF, U+A640-A69F, U+FE2E-FE2F;
}
/* carlito-cyrillic-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-cyrillic-400-normal.woff2) format("woff2"), url(./files/carlito-cyrillic-400-normal.woff) format("woff");
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}
/* carlito-greek-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-greek-ext-400-normal.woff2) format("woff2"), url(./files/carlito-greek-ext-400-normal.woff) format("woff");
  unicode-range: U+1F00-1FFF;
}
/* carlito-greek-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-greek-400-normal.woff2) format("woff2"), url(./files/carlito-greek-400-normal.woff) format("woff");
  unicode-range: U+0370-0377, U+037A-037F, U+0384-038A, U+038C, U+038E-03A1, U+03A3-03FF;
}
/* carlito-vietnamese-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-vietnamese-400-normal.woff2) format("woff2"), url(./files/carlito-vietnamese-400-normal.woff) format("woff");
  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB;
}
/* carlito-latin-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-latin-ext-400-normal.woff2) format("woff2"), url(./files/carlito-latin-ext-400-normal.woff) format("woff");
  unicode-range: U+0100-02BA, U+02BD-02C5, U+02C7-02CC, U+02CE-02D7, U+02DD-02FF, U+0304, U+0308, U+0329, U+1D00-1DBF, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF;
}
/* carlito-latin-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-latin-400-normal.woff2) format("woff2"), url(./files/carlito-latin-400-normal.woff) format("woff");
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}"
`;

exports[`scss imports > should import with default path 1`] = `
"/* carlito-cyrillic-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-cyrillic-ext-400-normal.woff2) format("woff2"), url(./files/carlito-cyrillic-ext-400-normal.woff) format("woff");
  unicode-range: U+0460-052F, U+1C80-1C8A, U+20B4, U+2DE0-2DFF, U+A640-A69F, U+FE2E-FE2F;
}
/* carlito-cyrillic-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-cyrillic-400-normal.woff2) format("woff2"), url(./files/carlito-cyrillic-400-normal.woff) format("woff");
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}
/* carlito-greek-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-greek-ext-400-normal.woff2) format("woff2"), url(./files/carlito-greek-ext-400-normal.woff) format("woff");
  unicode-range: U+1F00-1FFF;
}
/* carlito-greek-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-greek-400-normal.woff2) format("woff2"), url(./files/carlito-greek-400-normal.woff) format("woff");
  unicode-range: U+0370-0377, U+037A-037F, U+0384-038A, U+038C, U+038E-03A1, U+03A3-03FF;
}
/* carlito-vietnamese-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-vietnamese-400-normal.woff2) format("woff2"), url(./files/carlito-vietnamese-400-normal.woff) format("woff");
  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB;
}
/* carlito-latin-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-latin-ext-400-normal.woff2) format("woff2"), url(./files/carlito-latin-ext-400-normal.woff) format("woff");
  unicode-range: U+0100-02BA, U+02BD-02C5, U+02C7-02CC, U+02CE-02D7, U+02DD-02FF, U+0304, U+0308, U+0329, U+1D00-1DBF, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF;
}
/* carlito-latin-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-latin-400-normal.woff2) format("woff2"), url(./files/carlito-latin-400-normal.woff) format("woff");
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}"
`;

exports[`scss imports > should import with weight path and extension 1`] = `
"/* carlito-cyrillic-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-cyrillic-ext-400-normal.woff2) format("woff2"), url(./files/carlito-cyrillic-ext-400-normal.woff) format("woff");
  unicode-range: U+0460-052F, U+1C80-1C8A, U+20B4, U+2DE0-2DFF, U+A640-A69F, U+FE2E-FE2F;
}
/* carlito-cyrillic-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-cyrillic-400-normal.woff2) format("woff2"), url(./files/carlito-cyrillic-400-normal.woff) format("woff");
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}
/* carlito-greek-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-greek-ext-400-normal.woff2) format("woff2"), url(./files/carlito-greek-ext-400-normal.woff) format("woff");
  unicode-range: U+1F00-1FFF;
}
/* carlito-greek-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-greek-400-normal.woff2) format("woff2"), url(./files/carlito-greek-400-normal.woff) format("woff");
  unicode-range: U+0370-0377, U+037A-037F, U+0384-038A, U+038C, U+038E-03A1, U+03A3-03FF;
}
/* carlito-vietnamese-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-vietnamese-400-normal.woff2) format("woff2"), url(./files/carlito-vietnamese-400-normal.woff) format("woff");
  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB;
}
/* carlito-latin-ext-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-latin-ext-400-normal.woff2) format("woff2"), url(./files/carlito-latin-ext-400-normal.woff) format("woff");
  unicode-range: U+0100-02BA, U+02BD-02C5, U+02C7-02CC, U+02CE-02D7, U+02DD-02FF, U+0304, U+0308, U+0329, U+1D00-1DBF, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF;
}
/* carlito-latin-400-normal */
@font-face {
  font-family: "Carlito";
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/carlito-latin-400-normal.woff2) format("woff2"), url(./files/carlito-latin-400-normal.woff) format("woff");
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}"
`;
